packages/nx-openapi/src/generators/client/generator.spec.ts

```diff
@@ -0,0 +1,32 @@
+import { Tree, readProjectConfiguration } from '@nx/devkit';
+import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
+import { clientGenerator } from './generator';
+import { ClientGeneratorSchema } from './schema';
+
+describe('client generator', () => {
+    let tree: Tree;
+    beforeEach(() => {
+        tree = createTreeWithEmptyWorkspace();
+        tree.root = ''; // Trees root is automatically /virtual/ but that breaks the unit tests
```

Comment:
I believe this break unit tests because this is a virtual file tree but your unit test is trying to get file from a real file tree by `__dirname`. We may need to do more setup by mapping the root to the workspace folder.

Reply:
@ryanrixxh I found some issues with this generator which I think can be the problem that make it hard to do unit test. Let me know when you have some time to go over these with me.

Reply:
@ryanrixxh I have added 2 tickets for improvement. Let's work on those tickets and have separate PRs later.